File: packages/core/kube/dto/deployment.go
package dto

type DeploymentRef struct {
	Namespace string `json:"namespace"`
	Name      string `json:"name"`
}

type DeploymentCondition struct {
	Type               string
	Status             string
	Message            string
	Reason             string
	LastTransitionTime string
	LastUpdateTime     string
}

type Deployment struct {
	Name              string
	Namespace         string
	Pods              string
	Replicas          int32
	Age               string
	CreatedAt         string
	Labels            map[string]string
	Annotations       map[string]string
	ManagedFields     []ManagedField
	ReplicasDetail    string
	Selector          map[string]string
	NodeSelector      map[string]string
	StrategyType      string
	MaxSurge          string
	MaxUnavailable    string
	Conditions        []DeploymentCondition
	Tolerations       int
	TolerationDetails []TolerationDetail
	AffinityCount     int
	Affinities        string
}

type DeploymentSummary struct {
	Running int
	Pending int
}

File: packages/core/frontend/src/types/resources/deployment.ts
import type { ManagedField } from "./shared";
import type { TolerationDetail } from "./pod";

export interface DeploymentCondition {
  Type: string;
  Status: string;
  Message?: string;
  Reason?: string;
  LastTransitionTime?: string;
  LastUpdateTime?: string;
}

export interface Deployment {
  Name: string;
  Namespace: string;
  Pods: string;
  Replicas: number;
  Age: string;
  CreatedAt: string;
  Labels: Record<string, string>;
  Annotations: Record<string, string>;
  ManagedFields: ManagedField[];
  ReplicasDetail: string;
  Selector: Record<string, string>;
  NodeSelector: Record<string, string>;
  StrategyType: string;
  MaxSurge: string;
  MaxUnavailable: string;
  Conditions: DeploymentCondition[];
  Tolerations: number;
  TolerationDetails: TolerationDetail[];
  AffinityCount: number;
  Affinities: string;
}

export interface DeploymentSummary {
  Running: number;
  Pending: number;
}
